src/coreclr/tools/dotnet-pgo/Program.cs

```diff
@@ -1210,9 +1272,19 @@ static int InnerProcessTraceFileMain(CommandLineOptions commandLineOptions)
                                 continue;
                             if (codeViewData.Guid != e.ManagedPdbSignature)
                             {
-                                PrintError($"Dll mismatch between assembly located at \"{e.ModuleILPath}\" during trace collection and module \"{tsc.PEReaderToFilePath(ecmaModule.PEReader)}\"");
-                                mismatchErrors++;
-                                mismatch = true;
+                                if (modulesLoadedViaReference.Contains(ecmaModule) && duplicateModuleAnalysis[ecmaModule.Assembly.GetName().Name].Count > 1)
+                                {
+                                    // This is the case where a duplicate dll mismatch was avoided by specifying a -reference parameter
+                                    PrintMessage($"Disabling load of assembly data from assembly located at \"{e.ModuleILPath}\" during trace collection as module \"{tsc.PEReaderToFilePath(ecmaModule.PEReader)}\" is preferred, and does not match");
+                                    idParser.RemoveModuleIDFromLoader(e.ModuleID);
+                                    mismatchHandled = true;
```

Comment:
So essentially this allows dotnet-pgo to work even if multiple assemblies with the same name were loaded, by specifying one of them explicitly in `--reference`?

Reply:
Yes. Its not ideal, and results in a giant pile of warnings about being unable load load types, but it does work.